api/src/main/java/org/pmiops/workbench/api/ConceptSetsController.java

```diff
@@ -358,6 +360,63 @@ private void addConceptsToSet(
     return ResponseEntity.ok(toClientConceptSet(dbConceptSet));
   }
 
+  @Override
+  public ResponseEntity<ConceptSet> copyConceptSet(
+      String fromWorkspaceNamespace,
+      String fromWorkspaceId,
+      String fromConceptSetId,
+      CopyRequest copyRequest) {
+    Timestamp now = new Timestamp(clock.instant().toEpochMilli());
+    workspaceService.enforceWorkspaceAccessLevel(
+        fromWorkspaceNamespace, fromWorkspaceId, WorkspaceAccessLevel.READER);
+    Workspace toWorkspace =
+        workspaceService.get(
+            copyRequest.getToWorkspaceNamespace(), copyRequest.getToWorkspaceName());
+    Workspace fromWorkspace = workspaceService.get(fromWorkspaceNamespace, fromWorkspaceId);
+    workspaceService.getWorkspaceEnforceAccessLevelAndSetCdrVersion(
+        toWorkspace.getWorkspaceNamespace(),
+        toWorkspace.getFirecloudName(),
+        WorkspaceAccessLevel.WRITER);
+    if (toWorkspace.getCdrVersion().getCdrVersionId()
+        != fromWorkspace.getCdrVersion().getCdrVersionId()) {
+      throw new BadRequestException(
+          "New workspace does not have the same CDR version as current workspace");
+    }
+    org.pmiops.workbench.db.model.ConceptSet conceptSet =
+        conceptSetDao.findOne(Long.valueOf(fromConceptSetId));
+    org.pmiops.workbench.db.model.ConceptSet newConceptSet =
+        new org.pmiops.workbench.db.model.ConceptSet();
+    newConceptSet.setDomainEnum(conceptSet.getDomainEnum());
+    newConceptSet.setSurveysEnum(conceptSet.getSurveysEnum());
+
+    newConceptSet.setDescription(conceptSet.getDescription());
```

Comment:
Is there anywhere in the model where we need to keep up with the fact that this concept set was copied from the old one at this timestamp? Either for auditing internally or UX purposes? Any requirement to have a prefix for the title like "Copy of Concept Set 3"?

Reply:
None in the requirements specified. The name is user inputted, so we can't have a prefix of the title. (Since it could be arbitrary values and does not necessarily have the same name.)